sycl/doc/extensions/KernelProperties/KernelProperties.asciidoc

```diff
@@ -0,0 +1,525 @@
+
+  template <typename KernelName, int dimensions, typename PropertyList, typename... Rest>
+  void parallel_for(range<dimensions> numWorkItems,
+                    PropertyList properties,
+                    Rest&&... rest);
+
+  template <typename KernelName, int dimensions, typename PropertyList, typename... Rest>
+  void parallel_for(nd_range<dimensions> executionRange,
+                    PropertyList properties,
+                    Rest&&... rest);
+
+  template <typename KernelName, typename WorkgroupFunctionType, int dimensions, typename PropertyList>
+  void parallel_for_work_group(range<dimensions> numWorkGroups,
+                               PropertyList properties,
+                               const WorkgroupFunctionType &kernelFunc);
+
+  template <typename KernelName, typename WorkgroupFunctionType, int dimensions, typename PropertyList>
+  void parallel_for_work_group(range<dimensions> numWorkGroups,
+                               range<dimensions> workGroupSize,
+                               PropertyList properties,
+                               const WorkgroupFunctionType &kernelFunc);
+}
+}
+```
+
+Passing properties as an argument in this way allows properties to be
+associated with a kernel function without modifying its type. This enables
+the same kernel function (e.g. a lambda) to be submitted multiple times with
+different properties, or for libraries building on SYCL to add properties
+(e.g. for performance reasons) to user-provided kernel functions.
+
+All the properties defined in this extension have compile-time values. However,
+an implementation may support additional properties which could have run-time
+values. When this occurs, the `properties` parameter may be a `property_list`
+containing a mix of both run-time and compile-time values, and a SYCL
+implementation should respect both run-time and compile-time information when
+determining the correct way to launch a kernel. However, only compile-time
+information can modify the compilation of the kernel function itself.
+
+A simple example of using this extension to set a required work-group size
+and required sub-group size is given below:
+
+```c++
+sycl::ext::oneapi::property_list properties{sycl::ext::oneapi::work_group_size_v<8, 8>,
+                                            sycl::ext::oneapi::sub_group_size_v<8>};
+q.parallel_for(range<2>{16, 16}, properties, [=](id<2> i) {
+  a[i] = b[i] + c[i];
+}).wait();
+```
+
+== Encoding Properties into a Kernel
+
+In other situations it may be useful to encode a kernel's properties directly
+into its type, to ensure that a kernel cannot be launched without a property
+that it depends upon for correctness.
+
+To enable this use-case, this extension adds a mechanism for implementations to
+extract a property list from a kernel functor, if a kernel functor declares
+a `property_list` member variable. Note that this member variable must be
```

Comment:
Can we clarify what `property_list` means here? Name or type? If name then example below is not accurate - it uses `properties` as a name. If type then it's not clear how library can find it.

Reply:
Good point.  Clarified in e0c51887c69710c3a7995628c90b67a60f27de11.